service/controller/v5/resource/blobobject/update.go

```diff
@@ -0,0 +1,87 @@
+package blobobject
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/azure-operator/service/controller/v5/blobclient"
+	"github.com/giantswarm/azure-operator/service/controller/v5/controllercontext"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller"
+)
+
+func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateChange interface{}) error {
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	containerURL := cc.ContainerURL
+
+	containerObjectToUpdate, err := toContainerObjectState(updateChange)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	for _, containerObject := range containerObjectToUpdate {
+		if containerObject.Key != "" {
```

Comment:
I am wondering why we have to check this when we have something to iterate over. Is there any case where we append the item to the list while the key is empty?